File: hw1/district.py
import pandas as pd

district=pd.read_csv('data/districts.csv')
temp=district['municipality_info'].str[1:-1].str.split(',')
temp1=temp.str.get(0)
temp2=temp.str.get(1)
temp3=temp.str.get(2)
temp4=temp.str.get(3)

district['Population<500']=temp1
district['Population_500-1999']=temp2
district['Population_2000-9999']=temp3
district['Population>=10000']=temp4
district=district.drop(columns=['municipality_info'])

temp=district['unemployment_rate'].str[1:-1].str.split(',')
temp1=temp.str.get(0)
temp2=temp.str.get(1)
district['unemployment_rate_95']=temp1
district['unemployment_rate_96']=temp2
district=district.drop(columns=['unemployment_rate'])

temp=district['commited_crimes'].str[1:-1].str.split(',')
temp1=temp.str.get(0)
temp2=temp.str.get(1)
district['crime_rate_95']=temp1
district['crime_rate_96']=temp2
district=district.drop(columns=['commited_crimes'])

district.to_csv("district_py.csv",index=False)

File: hw1/district.R
library(rstudioapi)
setwd(dirname(rstudioapi::getActiveDocumentContext()$path))

library(reshape)
library(dplyr)
library(tidyr)
library(plyr)
library(stringr)
districts<-read.csv('data/districts.csv')
head(districts)

temp=str_split_fixed(str_sub(districts$municipality_info,2,-2),',',4)

districts['Population<500']=temp[,1]
districts['Population_500-1999']=temp[,2]
districts['Population_2000-9999']=temp[,3]
districts['Population>=10000']=temp[,4]
districts=select(districts,select=-c(municipality_info))

temp=str_split_fixed(str_sub(districts$unemployment_rate,2,-2),',',2)

districts['unemployment_rate_95']=temp[,1]
districts['unemployment_rate_96']=temp[,2]
districts=select(districts,select=-c(unemployment_rate))


temp=str_split_fixed(str_sub(districts$commited_crimes,2,-2),',',2)

districts['crime_rate_95']=temp[,1]
districts['crime_rate_96']=temp[,2]
districts=select(districts,select=-c(commited_crimes))

head(districts)

write.csv(districts,'district_r.csv',row.names=FALSE)
